data/simulation/trace.haskell.d.ts

```diff
@@ -4,85 +4,67 @@ export interface HaskellTraceEvent {
     event: HaskellEvent;
 }
 
-type MessageKind = "IB" | "EB" | "RB" | "VT";
+type BlockKind = "IB" | "EB" | "RB" | "VT";
+type BlockAction = "Generated" | "EnteredState";
 
 type HaskellEvent =
     | HaskellCpuEvent
-    | HaskellGeneratedEvent
-    | HaskellSentEvent
-    | HaskellReceivedEvent
-    | HaskellStateEvent;
+    | HaskellBlockEvent // Unified block event type
+    | HaskellNetworkEvent; // Combine Sent/Received into network events
 
 interface HaskellCpuEvent {
     tag: "Cpu";
-    node: number;
-    node_name: string;
-    duration_s: number;
+    node: string;
+    cpu_time_s: number;
     // CPU task types: Block validation (ValIB, ValEB, ValRB), Header validation (ValIH, ValRH), Vote validation (ValVote). Format: "<task_type>: <id>"
     task_label: string; // e.g., "ValIB: 6-29" or "ValRH: 6253064077348247640"
 }
 
-type HaskellGeneratedEvent =
-    | HaskellGeneratedIBEvent
-    | HaskellGeneratedEBEvent
-    | HaskellGeneratedRBEvent
-    | HaskellGeneratedVTEvent;
-
-interface HaskellGeneratedBaseEvent {
-    tag: "generated";
-    node: number;
-    node_name: string;
+// Base block event interface
+interface BaseBlockEvent {
+    type: `${BlockKind}${BlockAction}`;
+    node: string;
+    id: string;
     size_bytes: number;
 }
 
-interface HaskellGeneratedIBEvent extends HaskellGeneratedBaseEvent {
+interface InputBlockEvent extends BaseBlockEvent {
     kind: "IB";
-    id: string;
     slot: number;
     payload_bytes: number;
-    rb_ref: string;
+    rb_ref: string; // Reference to ranking block
 }
 
-interface HaskellGeneratedEBEvent extends HaskellGeneratedBaseEvent {
+interface EndorserBlockEvent extends BaseBlockEvent {
     kind: "EB";
-    id: string;
-    input_blocks: string[];
+    slot: number;
+    input_blocks: string[]; // References to input blocks
 }
 
-interface HaskellGeneratedRBEvent extends HaskellGeneratedBaseEvent {
+interface RankingBlockEvent extends BaseBlockEvent {
     kind: "RB";
-    id: string;
+    slot: number;
 }
 
-interface HaskellGeneratedVTEvent extends HaskellGeneratedBaseEvent {
+interface VoteEvent extends BaseBlockEvent {
     kind: "VT";
-    id: string;
     votes: number;
-    endorse_blocks: string[];
+    endorse_blocks: string[]; // References to endorser blocks
```

Comment:
I think we want `slot: number` here too, maybe it can go into BaseBlockEvent? Btw `slot` will be the one the slot is generated for.